webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,67 +20,57 @@ interface Props extends Partial<AnnouncementBarProps> {
     className?: string;
 }
 
-type State = {
-    dismissed: boolean;
-}
-
-export default class TextDismissableBar extends React.PureComponent<Props, State> {
-    constructor(props: Props) {
-        super(props);
-
-        this.state = {
-            dismissed: true,
-        };
-    }
+const TextDismissableBar = (props: Props) => {
+    const {allowDismissal, text, ...extraProps} = props;
+    const [dismissed, setDismissed] = useState<boolean>(false);
```

Comment:
We need to set the initial value to the correct value. If not, we may have a flash, since the first render will not render as dismissed, and then the value will be set at the effect, and will render with the new value.

But we may not want to call `localStorage.getItem` on every render, so we are going to use the function init of setState.

First, let's factor out of the component the logic to get the dimissed value:
```
const getDismissed = (text?: string) => localStorage.getItem(localStoragePrefix + text?.toString()) === 'true';
```
Then we init the state as this:
```
const [dismissed, setDismissed] = useState<boolean>(() => getDismissed(text));
```
Finally, we update the effect. Also, the current implementation of the effect only sets dismissed to true, and does not set it back to false if needed. We can fix it at the same time:
```
useEffect(() => {
    setDismissed(getDismissed(text));
}, [text]);
```
There is also a minor optimization to not call twice the `getDismissed` function in the first render by using `useDidUpdate` instead.